core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -690,6 +690,41 @@ sealed abstract class Chunk[+A] extends ChunkLike[A] { self =>
   def short(index: Int)(implicit ev: A <:< Short): Short =
     ev(apply(index))
 
+  /**
+   * Splits this chunk into `n` equally sized chunks.
+   */
+  final def split(n: Int): Chunk[Chunk[A]] = {
+    val length    = self.length
+    val quotient  = length / n
+    val remainder = length % n
+    val iterator  = self.iterator
+    val chunks    = ChunkBuilder.make[Chunk[A]]()
```

Comment:
May be faster to materialize into an array, and then copy portions of it into sub arrays. But this looks good for now!